tools/building.py

```diff
@@ -1491,6 +1496,8 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
   ret = check_call(cmd, stdout=stdout).stdout
   if outfile:
     save_intermediate(outfile, '%s.wasm' % tool)
+    global last_writing_binaryen_cmd
+    last_writing_binaryen_cmd = cmd
```

Comment:
I was hoping we could avoid this non-local stuff and keep all the logic in emcc.py.

Can't you add this after each possibly-stripping command:

```
if intermediate_debug_info == 0:
   did_strip = True
```

And then at then end you can just do:

```
# if we didn't already strip.. strip now
if not did_strip not debug_info:
  run_strip()
```

Reply:
The problem is that we don't issue all the commands here. For example, we call `building.minify_wasm_js`, which may run metadce, and which ends up doing multiple binaryen invocations, none of which emcc is aware of.

Reply:
But we pass `debug_info=intermediate_debug_info` to minify_wasm_js... right after that we can do:

```
if intermediate_debug_info == 0:
   did_strip = True
```

Reply:
Basically every time we use intermediate_debug_info we can check if it was zero right after we use it (or pass it out).

Reply:
If we ever use and its zero we know we did a strip, right?

Reply:
If we really do need a new global variable how about something more specific like `building.did_strip_binary` rather than `building.last_writing_binaryen_cmd`?

Reply:
Yes, we'd need to start passing it in, and receiving it back out, from all the building methods here. That seems more intrusive to me.

Another option is to always run `wasm-opt` to strip the names section, unconditionally. Shorter code, but significantly more work.

Yet another option might be to parse the wasm at the end to see if there is a names section there. Maybe doing it in python, where we only scan sections names, can be fast enough?

Reply:
ping @sbc100 - thoughts on moving forward here?

Reply:
I can hold my nose regarding the global state..   can you convert to a single `building.did_strip_binary` boolean rather than caching the last command line and then searching it?   For some reason it seems clearer to just have single bit of global state.

Reply:
Sounds good, done.